v2/dtos/addressable.go

```diff
@@ -0,0 +1,98 @@
+package dtos
+
+import (
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/models"
+)
+
+type Addressable struct {
+	common.Versionable `json:",inline"`
+
+	RESTAddressable    `json:",inline"`
+	MqttPubAddressable `json:",inline"`
+}
+
+type BaseAddressable struct {
+	Type string `json:"type" validate:"oneof='REST' 'MQTT'"`
+
+	Host string `json:"host"`
```

Comment:
Should also include `Protocol` here.

Reply:
"type" here should be able to map "protocol" constants automatically.

Reply:
Not 100% since MQTT can be TCP or TCPS, and REST can be HTTP or HTTPS. I am fine if you want to use a different property like `UseTls` or `EnabledTls` or something rather than `Protocol`

Reply:
Since we don't support TLS in the initial support-notifications and support-scheduler, may we just use "MQTT" and "REST" for now?  In the future, we can add "MQTT-TLS" and "REST-TLS" options to the type.  The new `EnableTLS` and other security related fields, such as `secretPath`, can be added in the future, too.
I thought `Protocol` is redundant.

Reply:
OK, I see you point. :-)